File: DarkOrbit 10.0/Net/netty/commands/AvailableModulesCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class AvailableModulesCommand
    {
        public static short ID = 1421;

        public List<StationModuleModule> modules;

        public AvailableModulesCommand(List<StationModuleModule> modules)
        {
            this.modules = modules;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeShort(30442);
            param1.writeInt(this.modules.Count);
            foreach(var module in this.modules)
            {
                param1.write(module.write());
            }
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/AvailableModulesCommand.java
package simulator.netty.serverCommands;


import java.io.DataOutputStream;
import java.io.IOException;
import java.util.Vector;

/**
 Created by Pedro on 22-03-2015.
 */
public class AvailableModulesCommand {

    public AvailableModulesCommand(Vector<StationModuleModule> param1) {
        this.modules = param1;
    }

    public static int ID = 4301;

    public Vector<StationModuleModule> modules;

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeInt(this.modules.size());
            for (StationModuleModule c : this.modules) {
                c.write(param1);
            }
        } catch (IOException e) {
        }
    }
}
